### PR TITLE
update workflow to support aqm.v7.0.8 implementation

### PR DESCRIPTION
**Pull request by JianpingHuang-NOAA**
## DESCRIPTION OF CHANGES: 
1) Update the Regional hourly Advanced Baseline Imager (ABI) and Visible Infrared Imaging Radiometer Suite (VIIRS) Emissions (RAVE) input datasets from version 1.3 to version 2.0, which use observations from the National Oceanic and Atmospheric Administration (NOAA)’s latest generation of U.S. polar-orbiting satellites (NOAA-21)
2) Update the list of AirNow’s observational surface sites for ozone (O3) and fine particulate matter (PM2.5) to improve bias-corrected products
3) Mitigate overprediction of fugitive dust
4) Increase accuracy of predicted dry deposition velocities for small gravitational settling and aerodynamic resistance values
Implement bug fixes and other code improvements

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] derecho.intel
- [ ] gaea.intel
- [ ] hera.gnu
- [ ] hera.intel
- [ ] hercules.intel
- [ ] jet.intel
- [ ] orion.intel
- [x] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

## CONTRIBUTORS (optional): 
Jianping Huang,  
Raffaele Montuoro 
Kai Wang
Brain Curtis
Lin Gan
Ho-Chun Huang, and others from AQ team at EMC
Collaborators from NESDIS and ARL

### CODE REVIEW

File: scripts/exaqm_bias_correction_o3.sh [hunk 1]
```diff
@@ -145,8 +145,8 @@ mkdir -p "${DATA}/data"
     mkdir -p "${cvt_input_dir}/${cvt_yyyy}/${cvt_pdy}"
     mkdir -p "${cvt_output_dir}/${cvt_yyyy}/${cvt_pdy}"
 
-    if [ "$(ls -A ${DCOMINairnow}/${cvt_pdy}/airnow)" ]; then
-      cpreq ${DCOMINairnow}/${cvt_pdy}/airnow/HourlyAQObs_${cvt_pdy}*.dat "${cvt_input_dir}/${cvt_yyyy}/${cvt_pdy}"
+    if [ -s ${DCOMINairnow}/${cvt_pdy}/airnow/HourlyAQObs_${cvt_pdy}00.dat ]; then
+      cp ${DCOMINairnow}/${cvt_pdy}/airnow/HourlyAQObs_${cvt_pdy}*.dat "${cvt_input_dir}/${cvt_yyyy}/${cvt_pdy}"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_bias_correction_o3.sh]: `cpreq` seemed to be the command, just confirming you intended to use just `cp` here

File: scripts/exaqm_bias_correction_o3.sh [hunk 2]
```diff
@@ -208,28 +209,53 @@ export err=$?; err_chk
...
+    target_dir="${COMOUTbicor}/bcdata.${!yyyymm_m}/airnow/csv/${!yyyy_m}/${!PDYm}"
+        
+    if [ ! -d "$target_dir" ]; then
+         mkdir -p "$target_dir"
+    fi
+
+    # Loop over each file individually
+    for file in "${DATA}/data/bcdata.${!yyyymm_m}/airnow/csv/${!yyyy_m}/${!PDYm}/HourlyAQObs_${!PDYm}"*.dat; do
+        if [ -s "$file" ]; then
+           cp "$file" "$target_dir"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_bias_correction_o3.sh]: and here

File: scripts/exaqm_bias_correction_o3.sh [hunk 3]
```diff
@@ -208,28 +209,53 @@ export err=$?; err_chk
...
+
+    target_dir="${COMOUTbicor}/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}"
+        
+    if [ ! -d "$target_dir" ]; then
+         mkdir -p "$target_dir"
+    fi
+
+    # Check if the file exists before attempting to copy it
+    if [ -s "${DATA}/data/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}/HourlyAQObs.${!PDYm}.nc" ]; then
+     cp "${DATA}/data/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}/HourlyAQObs.${!PDYm}.nc" "${COMOUTbicor}/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_bias_correction_o3.sh]: and here

File: scripts/exaqm_bias_correction_pm25.sh [hunk 4]
```diff
@@ -143,8 +143,9 @@ mkdir -p "${DATA}/data"
 
     mkdir -p "${cvt_input_dir}/${cvt_yyyy}/${cvt_pdy}"
     mkdir -p "${cvt_output_dir}/${cvt_yyyy}/${cvt_pdy}"
-    if [ "$(ls -A ${DCOMINairnow}/${cvt_pdy}/airnow)" ]; then
-      cpreq ${DCOMINairnow}/${cvt_pdy}/airnow/HourlyAQObs_${cvt_pdy}*.dat "${cvt_input_dir}/${cvt_yyyy}/${cvt_pdy}"
+    #jp if [ "$(ls -A ${DCOMINairnow}/${cvt_pdy}/airnow)" ]; then
+    if [ -s ${DCOMINairnow}/${cvt_pdy}/airnow/HourlyAQObs_${cvt_pdy}00.dat ]; then
+      cp ${DCOMINairnow}/${cvt_pdy}/airnow/HourlyAQObs_${cvt_pdy}*.dat "${cvt_input_dir}/${cvt_yyyy}/${cvt_pdy}"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_bias_correction_pm25.sh]: cp or cpreq here?

File: scripts/exaqm_bias_correction_pm25.sh [hunk 5]
```diff
@@ -209,8 +211,50 @@ fi
...
+
+      target_dir="${COMOUTbicor}/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}"
+
+      if [ ! -d "$target_dir" ]; then
+           mkdir -p "$target_dir"
+      fi
+
+      # Check if the file exists before attempting to copy it
+      if [ -s "${DATA}/data/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}/HourlyAQObs.${!PDYm}.nc" ]; then
+          cp "${DATA}/data/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}/HourlyAQObs.${!PDYm}.nc" "${COMOUTbicor}/bcdata.${!yyyymm_m}/airnow/netcdf/${!yyyy_m}/${!PDYm}"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_bias_correction_pm25.sh]: cp/cpreq?

File: scripts/exaqm_bias_correction_pm25.sh [hunk 6]
```diff
@@ -209,8 +211,50 @@ fi
...
+       else
+           message_warning="WARNING: File not found: HourlyAQObs.${!PDYm}.nc"
+           print_info_msg "${message_warning}"
+       fi
+    done
+
+   mkdir -p  ${COMOUTbicor}/bcdata.${yyyymm}/interpolated/pm25/${yyyy}
+   cp ${DATA}/out/pm25/${yyyy}/*nc ${COMOUTbicor}/bcdata.${yyyymm}/interpolated/pm25/${yyyy}
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_bias_correction_pm25.sh]: cp/cpreq?

File: scripts/exaqm_post_stat_o3.sh [hunk 7]
```diff
@@ -145,15 +145,29 @@ if [ "${cyc}" = "06" ] || [ "${cyc}" = "12" ]; then
...
 for grid in 227 196 198;do
   gg="grid${grid}"
-  wgrib2 ${NET}.${cycle}.ave_1hr_o3.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds earth -new_grid ${!gg} ${NET}.${cycle}.ave_1hr_o3.${grid}.grib2
+  wgrib2 ${NET}.${cycle}.ave_1hr_o3.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds grid -new_grid ${!gg} ${NET}.${cycle}.tmp.ave_1hr_o3.${grid}.grib2
+
+  # fix res flags
+  if [ "$grid" == "198" ] || [ "$grid" == "227" ]; then 
+     wgrib2 -set_flag_table_3.3 8 "${NET}.${cycle}.tmp.ave_1hr_o3.${grid}.grib2" -grib "${NET}.${cycle}.ave_1hr_o3.${grid}.grib2"
+  else
+     cp "${NET}.${cycle}.tmp.ave_1hr_o3.${grid}.grib2" "${NET}.${cycle}.ave_1hr_o3.${grid}.grib2"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_post_stat_o3.sh]: cp/cpreq?

File: scripts/exaqm_post_stat_o3.sh [hunk 8]
```diff
@@ -145,15 +145,29 @@ if [ "${cyc}" = "06" ] || [ "${cyc}" = "12" ]; then
...
 
   if [ "${cyc}" = "06" ] || [ "${cyc}" = "12" ]; then
-    wgrib2 ${NET}.${cycle}.ave_8hr_o3.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds earth -new_grid ${!gg} ${NET}.${cycle}.ave_8hr_o3.${grid}.grib2
+    wgrib2 ${NET}.${cycle}.ave_8hr_o3.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds grid -new_grid ${!gg} ${NET}.${cycle}.tmp.ave_8hr_o3.${grid}.grib2
+
+    # fix res flags
+    if [ "$grid" == "198" ] || [ "$grid" == "227" ]; then 
+        wgrib2 -set_flag_table_3.3 8 "${NET}.${cycle}.tmp.ave_8hr_o3.${grid}.grib2" -grib "${NET}.${cycle}.ave_8hr_o3.${grid}.grib2"
+    else
+        cp "${NET}.${cycle}.tmp.ave_8hr_o3.${grid}.grib2" "${NET}.${cycle}.ave_8hr_o3.${grid}.grib2"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_post_stat_o3.sh]: cp/cpreq?

File: scripts/exaqm_post_stat_o3.sh [hunk 9]
```diff
@@ -270,13 +284,22 @@ EOF1
...
+    wgrib2 ${NET}.${cycle}.max_8hr_o3.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds grid -new_grid ${!gg} ${NET}.${cycle}.tmp.max_8hr_o3.${grid}.grib2
+    wgrib2 ${NET}.${cycle}.max_1hr_o3.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds grid -new_grid ${!gg} ${NET}.${cycle}.tmp.max_1hr_o3.${grid}.grib2
+
+  # fix res flags
+  if [ "$grid" == "198" ] || [ "$grid" == "227" ]; then 
+     wgrib2 -set_flag_table_3.3 8 "${NET}.${cycle}.tmp.max_8hr_o3.${grid}.grib2" -grib "${NET}.${cycle}.max_8hr_o3.${grid}.grib2"
+     wgrib2 -set_flag_table_3.3 8 "${NET}.${cycle}.tmp.max_1hr_o3.${grid}.grib2" -grib "${NET}.${cycle}.max_1hr_o3.${grid}.grib2"
+  else
+     cp "${NET}.${cycle}.tmp.max_8hr_o3.${grid}.grib2" "${NET}.${cycle}.max_8hr_o3.${grid}.grib2"
+     cp "${NET}.${cycle}.tmp.max_1hr_o3.${grid}.grib2" "${NET}.${cycle}.max_1hr_o3.${grid}.grib2"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_post_stat_o3.sh]: cp/cpreq?

File: scripts/exaqm_post_stat_pm25.sh [hunk 10]
```diff
@@ -245,13 +254,22 @@ EOF1
...
+    wgrib2 ${NET}.${cycle}.ave_24hr_pm25.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds grid -new_grid ${!gg} ${NET}.${cycle}.tmp.ave_24hr_pm25.${grid}.grib2
+    wgrib2 ${NET}.${cycle}.max_1hr_pm25.${id_domain}.grib2 -set_grib_type c3b -new_grid_winds grid -new_grid ${!gg} ${NET}.${cycle}.tmp.max_1hr_pm25.${grid}.grib2
+
+    # fix res flags
+    if [ "$grid" == "198" ] || [ "$grid" == "227" ]; then 
+       wgrib2 -set_flag_table_3.3 8 "${NET}.${cycle}.tmp.ave_24hr_pm25.${grid}.grib2" -grib "${NET}.${cycle}.ave_24hr_pm25.${grid}.grib2"
+       wgrib2 -set_flag_table_3.3 8 "${NET}.${cycle}.tmp.max_1hr_pm25.${grid}.grib2" -grib "${NET}.${cycle}.max_1hr_pm25.${grid}.grib2"
+    else
+       cp "${NET}.${cycle}.tmp.ave_24hr_pm25.${grid}.grib2" "${NET}.${cycle}.ave_24hr_pm25.${grid}.grib2"
+       cp "${NET}.${cycle}.tmp.max_1hr_pm25.${grid}.grib2" "${NET}.${cycle}.max_1hr_pm25.${grid}.grib2"
```
Comment (BrianCurtis-NOAA) [inline scripts/exaqm_post_stat_pm25.sh]: cp/cpreq?

**Review by BrianCurtis-NOAA — CHANGES_REQUESTED** [review]
I've marked all the `cp` thinking you mentioned they all should be `cpreq` so you can either ignore them all or change where I marked. otherwise looks OK.

**Comment by BrianCurtis-NOAA** [comment]
Also there are conflicts to be resolved it says.

File: scripts/exaqm_bias_correction_o3.sh (hunk 1 above)
Comment (JianpingHuang-NOAA) [reply scripts/exaqm_bias_correction_o3.sh]: Yes, it is the one recommended by NCO

File: scripts/exaqm_bias_correction_pm25.sh (hunk 4 above)
Comment (JianpingHuang-NOAA) [reply scripts/exaqm_bias_correction_pm25.sh]: cpreq

Comment (JianpingHuang-NOAA) [reply scripts/exaqm_bias_correction_pm25.sh]: This is the same as the current operation job, exaqm_bias_correction_o3.sh at /lfs/h1/ops/prod
/packages/aqm.v7.0.7/scripts/

File: scripts/exaqm_bias_correction_pm25.sh (hunk 5 above)
Comment (JianpingHuang-NOAA) [reply scripts/exaqm_bias_correction_pm25.sh]: It is the same as the operation job, exaqm_bias_correction_pm25.sh at /lfs/h1/ops/prod
/packages/aqm.v7.0.7/scripts/.

File: scripts/exaqm_post_stat_o3.sh (hunk 7 above)
Comment (JianpingHuang-NOAA) [reply scripts/exaqm_post_stat_o3.sh]: It is the same as the operational job, exaqm_post_stat_o3.sh at /lfs/h1/ops/prod
/packages/aqm.v7.0.7/scripts/

File: scripts/exaqm_post_stat_o3.sh (hunk 8 above)
Comment (JianpingHuang-NOAA) [reply scripts/exaqm_post_stat_o3.sh]: It is the same as the operational job, exaqm_post_stat_o3.sh at /lfs/h1/ops/prod
/packages/aqm.v7.0.7/scripts/

File: scripts/exaqm_post_stat_o3.sh (hunk 9 above)
Comment (JianpingHuang-NOAA) [reply scripts/exaqm_post_stat_o3.sh]: It is the same as the operational job, exaqm_post_stat_o3.sh at /lfs/h1/ops/prod
/packages/aqm.v7.0.7/scripts/

File: scripts/exaqm_post_stat_pm25.sh (hunk 10 above)
Comment (JianpingHuang-NOAA) [reply scripts/exaqm_post_stat_pm25.sh]: It is the same as the operational job, exaqm_post_stat_pm25.sh at /lfs/h1/ops/prod
/packages/aqm.v7.0.7/scripts/

**Comment by BrianCurtis-NOAA** [comment]
OK. Seems all is good as is. @JianpingHuang-NOAA states it's tested successfully.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**